### DISCUSSION TITLE
Validation Form

### DISCUSSION THREAD

**Discussion by starmomou** [Budibase/budibase#3285] [Help] [answered]
Hi everyone,
I'm developing a simple web app. The app should be able to validate the input data. I can only use handlebars {{date}} to generate date. Maybe these are silly questions but there are little information related to the topic of handling the input data.
Any related material or information would be appreciated!! 
 
<img width="1165" alt="questions" src="https://user-images.githubusercontent.com/73826858/140725139-87a93dcd-7d9f-4a62-b900-9a41768ec9d9.png">

   ### My Problem
1. How to perform code If I want to use javascript to return **Date** instead of using handlebars?
2. How to perform code or use handlebars (not using regex in _Configure Validation_) etc. to check if user enters only number(int) in **Requested's ID**?
3. How to ensure the user has chosen at least one checkbox in **Items of Change** field
4. Will the feature of electronic signature be available in later releases?

**Comment by mjashanks** [comment, marked as answer]
Hi There.

**1. How to perform code If I want to use javascript to return Date instead of using handlebars?**

the javascript would be `return new Date();`
This might help, if you want to display in a certain format: https://stackoverflow.com/questions/3552461/how-to-format-a-javascript-date

**2. How to perform code or use handlebars (not using regex in Configure Validation) etc. to check if user enters only number(int) in Requested's ID?**

If you use a `Number Field` component, it should be impossible to type a number.

![image](https://user-images.githubusercontent.com/3524181/140750275-e2f88539-2932-4cbf-bed8-12b144557564.png)

**3. How to ensure the user has chosen at least one checkbox in Items of Change field**

don't think it's possible with a checkbox, but you can using a `Multi-Select Picker`. If using Budibase internal db, you would use a column of type `Multi-Select`:

![image](https://user-images.githubusercontent.com/3524181/140751530-561df9ad-4275-4297-8faf-874c49ad147b.png)

On the front-end, the multi-select picker looks like:

![image](https://user-images.githubusercontent.com/3524181/140751761-de8941a8-4b13-4692-80dc-ee711257bd09.png)

**4. Will the feature of electronic signature be available in later releases?**

We've been asked for this before - so yes, though I'm not sure when. Most people just need a canvas, i.e. where users can free-draw their signature on-screen. Is this enough? Or are you talking more about DocuSign or similar?

**Reply by starmomou** [reply]
Thank you for spending time answering my questions with patience! I have learnt so much skills from you. Yes, I simply want a canvas for users to sign their names  :D

**Reply by starmomou** [reply]
hi @mjashanks,

Could you show me the details on how to perform multi-select picker on the front-end because I can only add options by _Define Options_ and the **Items Of Change** cannot be added to the table column. I think the options should be provided by the data provider. Is there any settings should be done inside the form? I have added the column **Items Of Change** in my **Staff Info** table but still fails.

<img width="1223" alt="螢幕截圖 2021-11-09 下午3 33 05" src="https://user-images.githubusercontent.com/73826858/140881601-d2bccd8d-501d-4e8f-897f-387f44dc8ab3.png">

------------------------------------------------------------------------------------------------------------
<img width="910" alt="螢幕截圖 2021-11-09 下午4 08 15" src="https://user-images.githubusercontent.com/73826858/140886061-a10d9572-ea7d-4f25-9a69-851f7820b7d6.png">

**Reply by mjashanks** [reply]
Yes - change the "Options Source" to "Schema"
